avalanche/benchmarks/scenarios/continuous_task_agnostic_scenario.py

```diff
@@ -0,0 +1,330 @@
+        else:
+            probs[start:peak_start] = _get_linear_line(
+                start, peak_start, direction="up")
+        probs[peak_start:peak_end].add_(1)
+        if task_id == tasks - 1:
+            probs[peak_end:end].add_(1)
+        else:
+            probs[peak_end:end] = _get_linear_line(peak_end, end,
+                                                   direction="down")
+        return probs
+
+    # Total number of iterations
+    total_iters = len(original_stream) * iters_per_virtual_epoch
+
+    # Probabilities over all iterations (sub-experiences)
+    n_experiences = len(original_stream)
+    tasks_probs_over_iterations = [
+        _create_task_probs(total_iters, n_experiences, exp_id, beta=beta)
+        for exp_id in range(n_experiences)]
+
+    # Normalize probabilities
+    normalize_probs = torch.zeros_like(tasks_probs_over_iterations[0])
+    for probs in tasks_probs_over_iterations:
+        normalize_probs.add_(probs)
+    for probs in tasks_probs_over_iterations:
+        probs.div_(normalize_probs)
+    tasks_probs_over_iterations = torch.cat(
+        tasks_probs_over_iterations
+    ).view(-1, tasks_probs_over_iterations[0].shape[0])
+    tasks_probs_over_iterations_lst = []
+    for col in range(tasks_probs_over_iterations.shape[1]):
+        tasks_probs_over_iterations_lst.append(
+            tasks_probs_over_iterations[:, col])
+    tasks_probs_over_iterations = tasks_probs_over_iterations_lst
+
+    # Random cylic samplers over the datasets of all experiences in the stream
+    samplers = [iter(CyclicSampler(len(exp.dataset), shuffle=shuffle))
+                for exp in original_stream]
+
+    # The main iterator for the stream
+    def exps_iter():
+        for sub_exp_id in range(total_iters):
+            is_first_sub_exp = is_last_sub_exp = False
+            if sub_exp_id == 0:
+                is_first_sub_exp = True
+            if sub_exp_id == total_iters - 1:
+                is_last_sub_exp = True
+
+            n_samples = torch.Size([experience_size])
+            exp_per_sample_list = Categorical(
+                probs=tasks_probs_over_iterations[sub_exp_id]
+            ).sample(n_samples)
+
+            yield create_sub_exp_from_multi_exps(online_benchmark,
+                                                 original_stream,
+                                                 samplers,
+                                                 exp_per_sample_list,
+                                                 sub_exp_id,
+                                                 total_iters,
+                                                 is_first_sub_exp,
+                                                 is_last_sub_exp)
+
+    stream_name: str = getattr(original_stream, "name", "train")
+    return CLStream(
+        name=stream_name,
+        exps_iter=exps_iter(),
+        set_stream_info=True,
+        benchmark=online_benchmark,
+    )
+
+
+class ContinuousTaskAgnosticScenario(
```

Comment:
Do we need a new scenario? I think this is just an OCL scenario without task boundaries.